resource/cmake/CommonFunctions.cmake

```diff
@@ -0,0 +1,443 @@
+    set(CMAKE_C_STANDARD 99)
+    set(CMAKE_C_STANDARD_REQUIRED ON)
+    set(CMAKE_CXX_STANDARD 11)
+    set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+    # Set common variables
+    set(GATEWAY_DOC "${CMAKE_CURRENT_SOURCE_DIR}/doc")
+    set(COMMON_DIR "${CMAKE_CURRENT_SOURCE_DIR}/common")
+    set(JSON_PARSER_WRAPPER_DIR "${COMMON_DIR}/json_parser")
+    set(DDS_COMMON_DIR "${COMMON_DIR}/dds_specific")
+    set(TRANSFORMATION_COMMON_DIR "${COMMON_DIR}/transformation")
+    set(THIRD_PARTY_DIR "${CMAKE_CURRENT_SOURCE_DIR}/third-party")
+    set(LIBMODBUS_DIR "${THIRD_PARTY_DIR}/libmodbus")
+    set(PAHO_MQTT_C_DIR "${THIRD_PARTY_DIR}/paho.mqtt.c")
+    set(RESOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/resource")
+    set(UTILS_COMMON_DIR "${COMMON_DIR}/utils")
+    set(JSON_PARSER_DIR "${THIRD_PARTY_DIR}/json_parser")
+    set(ADAPTERS_DIR "${CMAKE_CURRENT_SOURCE_DIR}/plugins/adapters")
+    set(PROCESSORS_DIR "${CMAKE_CURRENT_SOURCE_DIR}/plugins/processors")
+    set(TRANSFORMATIONS_DIR "${CMAKE_CURRENT_SOURCE_DIR}/plugins/transformations")
+    set(EXAMPLES_DIR "${CMAKE_CURRENT_SOURCE_DIR}/examples")
+
+    set(STAGING_LIB_DIR "lib")
+    set(STAGING_BIN_DIR "bin")
+    set(STAGING_DOC_DIR "doc")
+    set(STAGING_TEST_DIR "test")
+    set(STAGING_EXAMPLES_DIR "examples")
+    set(STAGING_RESOURCE_DIR "resource")
+    set(STAGING_INCLUDE_DIR "include")
+
+    # If the RTIGATEWAY_ENABLE_${PluginName} is not specified, it will take the
+    # value of the RTIGATEWAY_ENABLE_ALL that is ON by default
+    option(RTIGATEWAY_ENABLE_ALL "Build everything" ON)
+    option(RTIGATEWAY_ENABLE_MODBUS "Build Modbus Adapter" ${RTIGATEWAY_ENABLE_ALL})
+    option(RTIGATEWAY_ENABLE_MQTT "Build MQTT Adapter" ${RTIGATEWAY_ENABLE_ALL})
+    option(RTIGATEWAY_ENABLE_FWD "Build Forwarder Processor" ${RTIGATEWAY_ENABLE_ALL})
+    option(RTIGATEWAY_ENABLE_TSFM_FIELD "Build Field Transformation" ${RTIGATEWAY_ENABLE_ALL})
+    option(RTIGATEWAY_ENABLE_TSFM_JSON "Build JSON Transformation" ${RTIGATEWAY_ENABLE_ALL})
+    option(RTIGATEWAY_ENABLE_TESTS "Build tester applications for enabled plugins" ${RTIGATEWAY_ENABLE_ALL})
+    option(RTIGATEWAY_ENABLE_EXAMPLES "Build examples applications for enabled plugins" ${RTIGATEWAY_ENABLE_ALL})
+    option(RTIGATEWAY_ENABLE_DOCS "Build documentation for enabled plugins" OFF)
+    option(RTIGATEWAY_ENABLE_SSL "Enable support for SSL/TLS" OFF)
+    option(RTIGATEWAY_ENABLE_LOG "Enable logging to stdout" OFF)
+    option(RTIGATEWAY_ENABLE_TRACE "Enable support for trace-level logging" OFF)
+
+    m_init_install_path()
+
+    # Set RTIGATEWAY_CONFIGURE_CONNEXT to check whether we need to configure
+    # RTI Connext DDS.
+    set(RTIGATEWAY_CONFIGURE_CONNEXT OFF)
+    if (RTIGATEWAY_ENABLE_MODBUS
+            OR RTIGATEWAY_ENABLE_MQTT
+            OR RTIGATEWAY_ENABLE_FWD
+            OR RTIGATEWAY_ENABLE_TSFM_FIELD
+            OR RTIGATEWAY_ENABLE_TSFM_JSON)
+        set(RTIGATEWAY_CONFIGURE_CONNEXT ON)
+    endif()
```

Comment:
I put that in case somebody wants to build only the documentation. Probably we should check `if (RTIGATEWAY_ENABLE_DOCS)`, or as you said, don't even support that use-case

Reply:
I would still load Connext every time, even if we just build the docs, to force users to have it installed on their systems.